tesseract_task_composer/core/include/tesseract_task_composer/core/task_composer_plugin_factory.h

```diff
@@ -93,15 +93,19 @@ class TaskComposerExecutorFactory
 
 class TaskComposerPluginFactory
 {
+private:
+  struct Implementation;
+  std::unique_ptr<Implementation> impl_;
+
```

Comment:
Why does this need to be moved to the top of the class?

Reply:
This may not be necessary I was fighting a "cannot delete incomplete type" error when I changed it. I will test without this change.

Reply:
This change was reverted.